### PR TITLE
CERT 7009 Add Chronicle Price Feed

### PR DESCRIPTION
**Pull request by nivcertora**
**📢 PR Summary: Integrate `ChronicleAPI` and Enhance Feed Price Verification**

**🔧 Key Changes:**

1. **API Module Restructuring:**
   - **Added `Quorum/apis/price_feeds/__init__.py`:**
     - Imported and exposed `ChainLinkAPI` and `ChronicleAPI` for centralized access.
     - Defined `__all__ = [ChainLinkAPI, ChronicleAPI]` for clear API exposure.
   - **Moved `ChainLinkAPI`:**
     - Renamed and relocated `Quorum/apis/chainlink_api.py` to `Quorum/apis/price_feeds/chainlink_api.py` for better organization within the `price_feeds` package.
  - **Moved `git_api`:**
     - relocated `Quorum/git_api` to `Quorum/apis/git_api` for better organization.
   - **Added `ChronicleAPI`:**
     - Introduced `Quorum/apis/price_feeds/chronicle_api.py` to interact with the Chronicle data feed API.
     - Implemented `ChronicleAPI` as a singleton to efficiently manage API requests and cache price feed data.

2. **Feed Price Verification Enhancements:**
   - **Updated `Quorum/checks/feed_price.py`:**
     - **Imported `ChainLinkAPI` and `ChronicleAPI`:** Adjusted import paths to the new `price_feeds` package.
     - **Initialized Both APIs:**
       - `chainlink_api = ChainLinkAPI()`
       - `chronicle_api = ChronicleAPI()`
     - **Aggregated Price Feeds from Multiple Sources:**
       - Fetched and stored price feeds from both Chainlink and Chronicle APIs into separate dictionaries for comprehensive verification.
     - **Enhanced Verification Logic:**
       - Extended verification to check price feed addresses against both Chainlink and Chronicle data sources.
       - Categorized addresses into verified and reported accordingly, providing more robust price feed validation.

3. **Utility Updates:**
   - **Updated `Quorum/utils/chain_enum.py`:**
     - **Renamed Blockchain Identifiers for Clarity and support chronicle name convention:**
       - `SCR` ➔ `SCROLL`
       - `ZK` ➔ `ZKSYNC`

**✅ Benefits:**

- **Comprehensive Feed Verification:** Integrating `ChronicleAPI` alongside `ChainLinkAPI` allows the tool to verify price feeds against multiple authoritative sources, enhancing reliability.
- **Improved Code Organization:** Structuring APIs within the `price_feeds` package promotes better maintainability and scalability.
- **Enhanced Verification Accuracy:** Dual-source verification reduces the risk of relying on a single data provider, ensuring more accurate and trustworthy price feed checks.

### CODE REVIEW

File: Quorum/apis/price_feeds/chronicle_api.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    Attributes:
+        session (requests.Session): A session object to manage HTTP requests.
+        memory (dict): A cache to store fetched price feed data for each chain.
+    """
+    
+    def __init__(self):
+        self.session = requests.Session()
+        self.memory = defaultdict(list)
+        self.info_url = "https://chroniclelabs.org/api/median/info/{}/{}/?testnet=false"
```
[the remark this answers is not in the document]
Comment (nivcertora) [reply Quorum/apis/price_feeds/chronicle_api.py:20]: Thanks!

File: Quorum/checks/feed_price.py [hunk 2]
```diff
@@ -29,21 +26,28 @@ def __init__(self, customer: str, chain: Chain, proposal_address: str, source_co
...
+        self.chainlink_api = ChainLinkAPI()
+        self.chronicle_api = ChronicleAPI()
+
         self.address_pattern = r'0x[a-fA-F0-9]{40}'
 
         # Retrieve price feeds from Chainlink API and map them by contract address
-        price_feeds = self.api.get_price_feeds_info(self.chain)
-        self.price_feeds_dict = {feed.contractAddress: feed for feed in price_feeds}
-        self.price_feeds_dict.update({feed.proxyAddress: feed for feed in price_feeds if feed.proxyAddress})
+        price_feeds = self.chainlink_api.get_price_feeds_info(self.chain)
```
[the remark this answers is not in the document]
Comment (nivcertora) [reply Quorum/checks/feed_price.py:35]: Thanks

File: Quorum/apis/price_feeds/chronicle_api.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    
+    def __init__(self):
+        self.session = requests.Session()
+        self.memory = defaultdict(list)
+        self.__pairs = self.__process_pairs()
+
+    def __process_pairs(self) -> dict[str, list[str]]:
+        pairs = requests.get(
+            "https://chroniclelabs.org/api/pairs?testnet=false"
+        ).json()
```
Comment (yoav-el-certora) [inline Quorum/apis/price_feeds/chronicle_api.py:25]: Any retries or raise_for_status needed here?

**Review by yoav-el-certora — APPROVED** [review]
Small comment for improvement but looks good

File: Quorum/apis/price_feeds/chronicle_api.py (hunk 3 above)
Comment (nivcertora) [reply Quorum/apis/price_feeds/chronicle_api.py]: Thanks added raise_for_status

**Pull request merged**